app/models/user.rb

```diff
@@ -318,9 +318,14 @@ def broken_personal_key?
     last_personal_key_at = self.encrypted_recovery_code_digest_generated_at
 
     if active_profile.present?
+      # This should only apply to encrypted_pii_recovery since encrypted_pii_recovery_multi_region
+      # was created after this time window.
       encrypted_pii_too_short =
-        active_profile.encrypted_pii_recovery.present? &&
-        active_profile.encrypted_pii_recovery.length < MINIMUM_LIKELY_ENCRYPTED_DATA_LENGTH
+        (active_profile.encrypted_pii_recovery_multi_region.present? &&
+        active_profile.encrypted_pii_recovery_multi_region.length <
+          MINIMUM_LIKELY_ENCRYPTED_DATA_LENGTH) ||
+        (active_profile.encrypted_pii_recovery.present? &&
+        active_profile.encrypted_pii_recovery.length < MINIMUM_LIKELY_ENCRYPTED_DATA_LENGTH)
```

Comment:
This is probably overly defensive since the potentially broken recovery PII should have been only been written to the single-region `encrypted_pii_recovery`. Any re-encryption to fix it would have written to both, so there should be no case where broken recovery PII is only present in the `encrypted_pii_recovery_multi_region` value. Because of that, it should be safe even in the 50/50 state.